.github/workflows/3rd-party-licenses.yaml

```
@@ -0,0 +1,107 @@
+# Run it locally with act
+#  1. Install act:
+#     `brew install act`
+#  2. Create a .secret file with the following content:
+#     `GITHUB_TOKEN=your_github_token`
+#  3. Create a pull_request.json file in case of a pull request with the following content:
+#     `{"pull_request": {"number": <PR number>, "head": {"ref": "<PR branch name>", "sha": "PR commit sha"}, "base": {"ref": "main"}}, "repository": {"name": "greenhouse", "owner": {"login": "cloudoperators"}}}`
+#  4. Run the following command:
+# `act pull_request --container-architecture linux/amd64 -P default=catthehacker/ubuntu:act-latest  -j check-3rd-party-licenses -e act_pull_request.json`
+
+name: 3rd Party License Checker
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  detect-changes:
+    runs-on: [default]
+    outputs:
+      changes: ${{ steps.packages-filters.outputs.changes }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Generate packages filters
+        run: |
+          for folder in ui/*; do
+            folder_name=$(basename "$folder")
+            echo "$folder_name: $folder/**" >> temp_packages_filters.yaml
+          done
+
+      - uses: dorny/paths-filter@v3
+        id: filters
+        with:
+          list-files: shell
+          filters: |
+            packages: ui/**
+
+      - uses: dorny/paths-filter@v3
+        id: packages-filters
+        with:
+          list-files: shell
+          filters: temp_packages_filters.yaml
+
+      - name: Show outputs
+        run: |
+          echo "====packages filter===="
+          cat temp_packages_filters.yaml
+          echo "========"
+          echo there are changes packages: ${{ steps.filters.outputs.packages}}
+          echo changed files: ${{ steps.filters.outputs.packages_files}}
+          echo package changes: ${{ steps.packages-filters.outputs.changes}}          
+          echo "===================="
+
+  check-3rd-party-licenses:
+    needs: detect-changes
+    runs-on: [default]
+    strategy:
+      matrix:
+        change: ${{fromJson(needs.detect-changes.outputs.changes)}}
```

I propose to move the node version here , to be able to test it in the future for multiple node version if we ever need it 

```suggestion
        change: ${{fromJson(needs.detect-changes.outputs.changes)}}
        node: [ 20.x ]
```

---

This is probably not needed, but it doesnt hurt... :)